node/resolvers/catalog/category.ts

```diff
@@ -1,72 +1,90 @@
-import { compose, find, last, path, prop, split } from 'ramda'
+import { compose, last, path, prop, split } from 'ramda'
 import { toIOMessage } from '../../utils/ioMessage'
+import { CatalogDataSource } from '../../dataSources/catalog'
 
-const lastSegment = compose<string, string[], string>(last, split('/'))
+const lastSegment = compose<string, string[], string>(
+  last,
+  split('/')
+)
 
 interface Category {
-  id: string,
-  url: string,
-  children: Category[],
+  id: string
+  url: string
+  name: string
+  children: Category[]
+}
+
+type CategoryMap = Record<string, Category>
+
+const appendToMap = (mapCategories: CategoryMap, category: Category) => {
+  mapCategories[category.id] = {
+    ...category,
+  }
+
+  mapCategories = category.children.reduce(appendToMap, mapCategories)
+
+  return mapCategories
+}
+
+const getCategoryInfo = async (catalog: CatalogDataSource, id: string) => {
+  const categories = (await catalog.categories(3)) as Category[]
```

Comment:
this looks a bit magical, could you comment on what this does/why this was done?